python_modules/libraries/dagster-embedded-elt/dagster_embedded_elt/sling/asset_decorator.py

```diff
@@ -0,0 +1,181 @@
+import re
+from typing import Any, Callable, Iterable, Mapping, Optional
+
+from dagster import (
+    AssetKey,
+    AssetsDefinition,
+    AssetSpec,
+    BackfillPolicy,
+    FreshnessPolicy,
+    PartitionsDefinition,
+    multi_asset,
+)
+from dagster._annotations import public
+from dagster._core.definitions.asset_dep import CoercibleToAssetDep
+from dagster._core.definitions.auto_materialize_policy import AutoMaterializePolicy
+
+from dagster_embedded_elt.sling.sling_replication import SlingReplicationParam, validate_replication
+
+
+def get_streams_from_replication(replication: Mapping[str, Any]) -> Iterable[str]:
+    """Returns the set of streams from a Sling replication config."""
+    keys = []
+
+    for key, value in replication["streams"].items():
+        if isinstance(value, dict):
+            asset_key_config = value.get("meta", {}).get("dagster", {}).get("asset_key", [])
+            keys.append(asset_key_config if asset_key_config else key)
+        else:
+            keys.append(key)
+    return keys
+
+
+class DagsterSlingTranslator:
+    @classmethod
+    def sanitize_stream_name(cls, stream_name: str) -> str:
+        """A function that takes a stream name from a Sling replication config and returns a
+        sanitized name for the stream.
+        By default, this removes any non-alphanumeric characters from the stream name and replaces
+        them with underscores, while removing any double quotes.
+        """
+        return re.sub(r"[^a-zA-Z0-9_.]", "_", stream_name.replace('"', ""))
+
+    @public
+    @classmethod
+    def get_asset_key_for_target(cls, stream_name: str, target_prefix: str = "target") -> AssetKey:
```

Comment:
Let's be consistent with `dagster-dbt`. Also, any reason why we shouldn't just pass in the entire stream object itself?

```suggestion
    def get_asset_key(cls, stream_config: Mapping[str, Any]) -> AssetKey:
```